src/nomad_simulations/schema_packages/outputs.py

```diff
@@ -163,8 +163,13 @@ def set_model_system_ref(self) -> Optional[ModelSystem]:
         """
         if self.m_parent is not None:
             model_systems = self.m_parent.model_system
-            if model_systems is not None and len(model_systems) == 1:
-                return model_systems[-1]
+            outputs = self.m_parent.outputs
+            if (
+                isinstance(model_systems, list)
+                and isinstance(outputs, list)
+                and len(model_systems) == len(outputs)
+            ):
```

Comment:
Note: this version is more restrictive (will fail with arrays, for example), but also more safe. Since `len` is a method of `list`, there's no reliance on duck typing here.